String resources, npes, declaration/casting redundancies refactor

gubatron opened the pull request:
- Silenced some trivial NPE warnings
 - Removed redundant casts and initializations, and respective import cleanups
 - Trivial lambda refactors

gubatron committed:
Refactored amount related strings to use resources (+more)

 - Silenced some trivial NPE warnings
 - Removed redundant casts and initializations
 - Trivial lambda refactors

gubatron committed:
import cleanup

gubatron committed:
extra space

gubatron commented:
The final goal with the string resource stuff is to be able to translate the app easily to multiple languages.

zquestz commented on app/src/main/java/cash/bchd/android_neutrino/MainActivity.java:
```diff
@@ -369,9 +368,11 @@ private void copyToClipboard(String data) {
-        snackbar.show();
+        if (clipboardManager != null) {
+            clipboardManager.setPrimaryClip(clipData);
+            Snackbar snackbar = Snackbar.make(mCLayout, "Address copied clipboard.", Snackbar.LENGTH_LONG);
```
Shouldn't this string be in the XML below?

gubatron replied:
yes

zquestz commented on app/src/main/java/cash/bchd/android_neutrino/SendActivity.java:
```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
+                boolean inputError = false;
+                boolean valid = wallet.validateAddress(address.getText().toString());
+                if (!valid) {
+                    layout.setError("INVALID BITCOIN CASH ADDRESS");
```
Shouldn't this string be in the XML below?

gubatron replied:
yes, to all your comments with strings, I just started with a couple.

If merged I can continue.

zquestz replied:
Excellent. Thanks for the good start!

zquestz commented on app/src/main/java/cash/bchd/android_neutrino/SendActivity.java:
```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
+                Amount toSpend;
+                if (inputAmount.getText().toString().equals("")) {
+                    TextInputEditText etAmount = findViewById(R.id.amountInput);
+                    etAmount.setError("INVALID AMOUNT");
```
And here.

zquestz commented on app/src/main/java/cash/bchd/android_neutrino/SendActivity.java:
```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
+                }
+                if (toSpend.getSatoshis() == 0 || toSpend.getSatoshis() > balance.getSatoshis()) {
+                    TextInputEditText etAmount = findViewById(R.id.amountInput);
+                    etAmount.setError("INVALID AMOUNT");
```
Another.

zquestz commented on app/src/main/java/cash/bchd/android_neutrino/SendActivity.java:
```diff
@@ -180,103 +162,98 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                 }
+                startActivity(intent1);
+            } catch (Exception e) {
+                Snackbar snackbar = Snackbar.make(sendLayout, "Error creating transaction.", Snackbar.LENGTH_LONG);
```
All the Snackbar strings as well?

gubatron committed:
New extracted string resources from Main/Receive/Send activities

gubatron committed:
MainActivity refactored to use new strings, cleanup

gubatron committed:
ReceiveActivity refactors

 - String resources
 - Lots of Null Pointer Exception checks
 - Lambda refactors
 - 2 condition switch into if
 - boolean simplification
 - same var asigned to itself removed
 - redundant casts
 - import cleanup

gubatron committed:
SendActivity refactors

Same as the ones performed for ReceiveActivity, string resources, NPEs, boolean refactors, import cleanup

gubatron commented:
All String resources in MainActivity, SendActivity, ReceiveActivity extracted.

Also performed a lot of minor cleanups and a bunch of NPE warnings have been taken care of, along with several others related to missuse of switch statements, boolean simplifications, anonymous classes turned into lambda expressions, same variable assignment, unused variables cleaned up, import cleanups.

Broke fixes into individual commits for easier reviewing / cherry-picking.

gubatron commented on app/src/main/java/cash/bchd/android_neutrino/ReceiveActivity.java:
```diff
@@ -247,8 +244,6 @@ private void displayQRPopup() {
             int width = point.x;
             int height = point.y;
             int smallerDimension = width < height ? width : height;
-            smallerDimension = smallerDimension;
```
🤣

zquestz commented:
This looks great, however will wait for Java folks to chime in. My Java foo needs work. =)
